README.md

```diff
@@ -1,227 +1,33 @@
-docker run -e uid="$(id -u)" -e gid="$(id -g)" -v $PWD:/work -w /work wasmsdk:v2 /build_wasm.sh
-```
-
-## Dependencies for building WASM modules:
-
-If you do not wish to use the Docker file, the dependencies can be installed by script (sdk\_container.sh), or by hand.
-
-### protobuf v3.9.1
-
-You must install the version of protobuf on your build system that matches the libprotobuf.a files (without any patches) so that the generated code matches the .a library.  Currently this is based on tag v3.9.1 of https://github.com/protocolbuffers/protobuf.
-
-```bash
-git clone https://github.com/protocolbuffers/protobuf
-cd protobuf
-git checkout v3.9.1
-git submodule update --init --recursive
-./autogen.sh
-./configure
-make
-make check
-sudo make install
-```
-
-### emscripten
-
-```bash
-git clone https://github.com/emscripten-core/emsdk.git
-cd emsdk
-./emsdk update-tags
-./emsdk install 3.1.7
-./emsdk activate 3.1.7
-
-source ./emsdk\_env.sh
-```
-
-It is possible later versions will work, e.g.
-
-```bash
-./emsdk update-tags
-./emsdk install latest
-./emsdk activate latest
-```
-
-However 3.1.7 is known to work.
-
-### Rebuilding the libprotobuf.a files
-
-If want to rebuild the libprotobuf.a files or use a different version see the instructions at https://github.com/kwonoj/protobuf-wasm. Commit 4bba8b2f38b5004f87489642b6ca4525ae72fe7f works for protobuf v3.9.x.
-
-```bash
-git clone https://github.com/protocolbuffers/protobuf protobuf-wasm
-cd protobuf-wasm
-git checkout v3.9.1
-git clone https://github.com/kwonoj/protobuf-wasm wasm-patches
-cd wasm-patches && git checkout 4bba8b2f38b5004f87489642b6ca4525ae72fe7f && cd ..
-git apply wasm-patches/*.patch
-./autogen.sh
-emconfigure ./configure --disable-shared CXXFLAGS="-O3 -flto"
-emmake make
-cd ..
-cp protobuf-wasm/src/.libs/libprotobuf-lite.a ${CPP_API}/libprotobuf-lite.a
-cp protobuf-wasm/src/.libs/libprotobuf.a ${CPP_API}/libprotobuf.a
-```
-
-### WAVM binaries
-
-```bash
-git clone git@github.com:WAVM/WAVM.git
-cd WAVM
-cmake "."
-make
-sudo make install
-```
-
-Note: ensure /usr/local/bin is in your path
-
+Proxy-Wasm is a specification and supporting framework for using
+[WebAssembly](https://webassembly.org) (Wasm) to extend the functionality of
+network proxies. It enables developers to write custom logic (plugins) that are
+compiled to Wasm modules and then loaded and executed by the proxy.
+
+Proxy-Wasm consists of multiple parts:
+
+* An [ABI](https://github.com/proxy-wasm/spec) that specifies the low-level
+  interface between network proxies and Wasm virtual machines that run the
+  plugins.
+* [Host implementations](https://github.com/proxy-wasm/spec#host-environments)
+  of the ABI, provided by network proxies.
+* [Language-specific SDKs](https://github.com/proxy-wasm/spec#sdks) that layer
+  on top of the ABI, providing a more natural and programmer-friendly API for
+  invoking and implementing Proxy-Wasm functions and callbacks.
+  
+This repository provides the C++ SDK.
```

Comment:
Worth mentioning here that this uses Emscripten to compile C to wasm? (and linking to https://emscripten.org/)

Reply:
That's an implementation detail, and honestly, it's not clear if that's something that provides any value at this point compared to using `clang --target=wasm32-wasi` (@mpw it might be worth filing an issue about that).

Reply:
The previous language and link for Emscripten wasn't removed, it was just moved to the building.md page, since it's more of a build/implementation detail.

Filed #167 for updating SDK to document using `clang --target=wasm32-wasi` directly.